server/borsen/io/feed_parsers/ritzau.py

```diff
@@ -42,8 +53,63 @@ def do_mapping(self, item, item_xml, setting_param_name=None, namespaces=None):
         if default_section:
             item.setdefault("anpa_category", []).append(default_section[0])
 
+        raw_headline = item.get("headline", "")
+        cleaned_headline = self.strip_suffix(raw_headline)
+        item["headline"] = cleaned_headline
+
+        # Step 1: Try matching by GUID
+        if item.get("guid"):
+            existing = self.find_by_guid(item["guid"])
+            if existing:
+                self.update_existing_item(item, existing)
+                return item
+
+        # Step 2: Match similar headlines (≥80%)
+        similar_article = self.find_similar_article(cleaned_headline)
+        if similar_article:
+            self.update_existing_item(item, similar_article)
+            return item
+
         return item
 
+    def find_by_guid(self, guid):
+        """Find item by GUID."""
+        try:
+            return superdesk.get_resource_service("ingest").find_one(req=None, guid=guid)
+        except Exception as e:
+            logger.warning(f"Error finding article by GUID {guid}: {str(e)}")
+            return None
+
+    def find_similar_article(self, headline):
+        """Find similar article by headline match in last 24 hours."""
+        if not headline:
+            return None
+
+        try:
+            cutoff = datetime.utcnow() - timedelta(hours=24)
+            ingest_items = superdesk.get_resource_service("ingest").get(
```

Comment:
would be good to do a preselection using [more_like_this](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/query-dsl-mlt-query.html) query

Reply:
actually if you set `minimum_should_match` to `80%` it might be enough